var/spack/repos/builtin/packages/intel-tbb/package.py

```diff
@@ -238,3 +238,21 @@ def libs(self):
         shared = True if '+shared' in self.spec else False
         return find_libraries(
             'libtbb*', root=self.prefix, shared=shared, recursive=True)
+
+    @run_after('install')
+    def install_pkgconfig(self):
+        libdir = self.spec['tbb'].libs.directories[0]
```

Comment:
The `spec['tbb']` looks a bit suspicous to me.

I assume, you want to reference the package itself.

* It should not be `'tbb'`, but `'intel-tbb'`?
* And even better, check into `self.spec.libs`?

If neither work, we probably need to understand, why. (And for the time being, we could accept the above form).

Reply:
`'tbb'` is a virtual package (for the moment, the only implementation is `intel-tbb` so this may be historical?)

Reply:
But looking at the implementation here `self.spec` would be better.

Reply:
> 'tbb' is a virtual package (for the moment, the only implementation is intel-tbb so this may be historical?

It seems that `intel-oneapi-tbb` also provides 'tbb' .
You can see the background by reading this PR  https://github.com/spack/spack/pull/25613  

> But looking at the implementation here self.spec would be better.

Unfortunately, there doesn't seem to be  `self.spec.libs`.
But it seems worthwhile (for example, reducing the risk of referencing other versions of intel-tbb), so I'll try to work on it.

Reply:
`self.libs.directories[0]` is working :+1:

Reply:
When I looked it up, there were some different styles in `def install_pkgconfig (self):` (e.g. leveldb, lmdb, nss, rocksdb, and snappy).

snappy's `install_pkgconfig` seems to be the best solution.
https://github.com/spack/spack/blob/b88c80bb1becf18dd7f53246299f78de84f99d2f/var/spack/repos/builtin/packages/snappy/package.py#L43 

I stop using self.libs.directories [0] and use self.prefix.lib.pkgconfig instead